src/web.ts

```diff
@@ -214,6 +195,170 @@ export async function app(probot: Probot, probotOptions: ApplicationFunctionOpti
 
     logger.info(`Marked the PR #${pr.number} as closed on ${organization}/${repoName} for the branch ${pr.base.ref}`);
   });
+
+  /**
+   * Handle a commit status or check run completing for a PR.
+   *
+   * This will merge the PR if this is the last required successful commit status or check run. This will create a
+   * GitHub issue if the commit status or check run failed.
+   *
+   * Note that a PR could get stuck if the required checks on the branch change while the PR CI is still running.
+   *
+   * @param {Octokit} client the GitHub client to use that is authenticated as the GitHub installation.
+   * @param {string} organization the GitHub organization of the repository that the PR belongs to.
+   * @param {string} repoName the GitHub repository name of the repository that the PR belongs to.
+   * @param {PullRequest | CheckRunPullRequest} pr the pull-request that the CI update is for.
+   * @param {string} checkName the name of the commit status or check run that just completed.
+   * @param {boolean} success determines if the commit status or check run was successful.
+   * @return {Promise<boolean>} a Promise that resolves to a boolean indicating if the PR was handled. This is false if
+   *   if the PR is blocked or isn't known to Magic Mirror.
+   */
+  const handlePRCIUpdate = async (
+    client: Octokit,
+    organization: string,
+    repoName: string,
+    pr: PullRequest | CheckRunPullRequest,
+    checkName: string,
+    success: boolean,
+  ): Promise<boolean> => {
+    const repo = await db.getOrCreateRepo(organization, repoName);
+    const pendingPR = await db.getPendingPRByPRID(repo, pr.number);
+    if (!pendingPR) {
+      logger.debug(`The PR #${pr.number} on ${organization}/${repoName} does not apply`);
+      return false;
+    }
+
+    if (pendingPR.action === PRAction.Blocked) {
```

Comment:
Would we want to check if the PR checks are passing before the PR is blocked? If a retest is needed (ex:  a PR in one repo contains a dependency on another repo and fails until a PR in the other repo is merged; or a check fails because of a prow issue, etc) then it seems to me that the PR will remain blocked until the issue is closed, even if all checks pass on a retest. If the checks fail and pass later I think it might make sense to let the PR through.

Reply:
It's a good point, but I think to simplify the workflow, I'd like to keep it as is. If we find this annoying, we can improve this behavior.

To clarify, with this current code, when the PR is blocked, that means that Magic Mirror has already created a GitHub issue indicating that the CI has failed.

The squad would then notice the GitHub issue. They could then retry the CI and manually merge the PR and then close the GitHub issue created by Magic Mirror. Once that has occurred, then Magic Mirror will start syncing again for the specific Stolostron repo's branch that was previously blocked.

Reply:
Thinking about it more, while manually closing the issue after triggering a retest on the forked PR might be a bit of a pain, I don't think that it should block this PR from going through. But I do think automatically closing the issue if a retest causes the required checks to pass could be a helpful feature